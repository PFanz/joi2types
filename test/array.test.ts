import Joi from '@hapi/joi';
import Joi2Types, { joi2JsonSchema } from '../src';

test('joi2Types array', async () => {
  const schema = Joi.array();
  expect(
    joi2JsonSchema(schema)
  ).toEqual({
    type: 'array',
  })
  const types = await Joi2Types(schema);
  expect(types.trim()).toMatchSnapshot();

  const schemaWithItems = Joi.array().items(Joi.string());
  expect(
    joi2JsonSchema(schemaWithItems)
  ).toEqual({
    type: 'array',
    items: {
      type: 'string',
    }
  })
  const typesWithItems = await Joi2Types(schemaWithItems);
  expect(typesWithItems.trim()).toMatchSnapshot();
})

test('joi2Types array object', async () => {
  const schema = Joi.array().items(Joi.object({
    path: Joi.string().description('Any valid URL path'),
    component: Joi.string().description('A React component to render only when the location matches.'),
    redirect: Joi.string().description('navigate to a new location'),
    exact: Joi.boolean().description('When true, the active class/style will only be applied if the location is matched exactly.'),
  }).unknown());
  expect(
    joi2JsonSchema(schema)
  ).toEqual({
    type: 'array',
    items: {
      type: 'object',
      properties: {
        path: {
          type: 'string',
          description: 'Any valid URL path'
        },
        component: {
          type: 'string',
          description: 'A React component to render only when the location matches.'
        },
        redirect: {
          type: 'string',
          description: 'navigate to a new location',
        },
        exact: {
          type: 'boolean',
          description: 'When true, the active class/style will only be applied if the location is matched exactly.'
        }
      }
    }
  })
})
