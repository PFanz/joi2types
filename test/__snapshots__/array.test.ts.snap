// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joi2Types array 1`] = `"export type JoiTypes = any[];"`;

exports[`joi2Types array 2`] = `"export type JoiTypes = string[];"`;

exports[`joi2Types array object 1`] = `
"export type JoiTypes = {
  /**
   * Any valid URL path
   */
  path?: string;
  /**
   * A React component to render only when the location matches.
   */
  component?: string;
  /**
   * navigate to a new location
   */
  redirect?: string;
  /**
   * When true, the active class/style will only be applied if the location is matched exactly.
   */
  exact?: boolean;
  routes?: any[];
  [k: string]: any;
}[];"
`;
